File: ProjectCS/StyledTXTReportExporter.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SessionResultsCS
{
    public class StyledTXTReportExporter : TXTReportExporter
    {
        // Стиль таблицы
        public class TableStyle
        {
            public char CrossSymbol { get; set; }
            public char HorizontalSymbol { get; set; }
            public char VerticalSymbol { get; set; }

            public TableStyle(char crossSymbol, char horizontalSymbol, char verticalSymbol)
            {
                CrossSymbol = crossSymbol;
                HorizontalSymbol = horizontalSymbol;
                VerticalSymbol = verticalSymbol;
            }
        }


        public TableStyle Style { get; set; }


        public StyledTXTReportExporter(string filePath, TableStyle style, string? title = null, string? body = null, bool includeDate = false)
            : base(filePath, title, body, includeDate)
        {
            Style = style;
        }

        public override string RenderTable(GroupTableData data)
        {
            int[] widths = CalculateColumnWidths(data);
            string separator = TableSeparator(widths, Style.CrossSymbol, Style.HorizontalSymbol);

            StringBuilder builder = new StringBuilder();

            builder.AppendLine(separator);
            builder.AppendLine(TableRow(data.TableHeader(), widths, Style.VerticalSymbol));
            builder.AppendLine(separator);
            foreach (string[] cells in data.TableBodyToStrings(true))
            {
                builder.AppendLine(TableRow(cells, widths, Style.VerticalSymbol));
            }
            builder.AppendLine(separator);
            builder.AppendLine(TableRow(data.TableDisciplineAverages(), widths, Style.VerticalSymbol));
            builder.Append(separator);

            return builder.ToString();
        }
    }
}

File: Project/StyledTXTReportExporter.cpp
#include "StyledTXTReportExporter.hpp"

#include <sstream>

StyledTXTReportExporter::StyledTXTReportExporter(const std::string& filePath, const TableStyle& style, const std::string& title, const std::string& body, bool includeDate)
	:TXTReportExporter(filePath, title, body, includeDate), m_tableStyle(style) {}

StyledTXTReportExporter::StyledTXTReportExporter(const StyledTXTReportExporter& other) :
	TXTReportExporter(other), m_tableStyle(other.m_tableStyle) {}


const TableStyle & StyledTXTReportExporter::GetTableStyle() const {
	return m_tableStyle;
}

void StyledTXTReportExporter::SetTableStyle(const TableStyle& style) {
	m_tableStyle = style;
}

std::string StyledTXTReportExporter::RenderTable(const GroupTableData& data) {
	std::vector<int> widths = CalculateColumnWidths(data);
	std::string separator = TableSeparator(widths, m_tableStyle.GetCrossSymbol(), m_tableStyle.GetHorizontalSymbol());

	std::stringstream ss;

	ss << separator << std::endl;
	ss << TableRow(data.TableHeader(), widths, m_tableStyle.GetVerticalSymbol()) << std::endl;
	ss << separator << std::endl;
	for (const std::vector<std::string> &row : data.TableBodyToStrings(true)) {
		ss << TableRow(row, widths, m_tableStyle.GetVerticalSymbol()) << std::endl;
	}
	ss << separator << std::endl;
	ss << TableRow(data.TableDisciplineAverages(), widths, m_tableStyle.GetVerticalSymbol()) << std::endl;
	ss << separator;

	return ss.str();
}
